source/Ship.cpp

```diff
@@ -2640,7 +2642,17 @@ int Ship::TakeDamage(const Projectile &projectile, bool isBlast)
 	else if(shieldDamage > shields)
 		shieldFraction = min(shieldFraction, shields / shieldDamage);
 	shields -= shieldDamage * shieldFraction;
-	hull -= hullDamage * (1. - shieldFraction);
+	
+	double hullFraction = (1. - shieldFraction);
+	double disabledFraction = 0;
+	double nonDisabledHull = hull - MinimumHull() + .25d;
+	if(hullDamage * hullFraction > nonDisabledHull)
+	{
+		hullFraction = max(nonDisabledHull/hullDamage,0.d);
+		disabledFraction = 1 - shieldFraction - hullFraction;
+	}
+	hull -= hullDamage * hullFraction + disabledDamage * disabledFraction;
```

Comment:
Can we also get the special case when disabledDamage is negative?

I think that would be something like:

```
hull -= hullDamage * hullFraction
// disabled damage might actually restore hulls. The max ensures that restores don't re-enable ships.
if (hull < (MinimumHull() - 0.25d))
  hull = max(hull - disabledDamage * disabledFraction, MinimumHull() - .25d)
```

Reply:
So if a weapon has negative disabled damage (healing) it should only restore it to max disabled health, and would require negative hull damage to heal past that?  I will take a look at that then seems like a good idea.

Reply:
This has opened up a new can of worms in determining the order of damage to be dealt.  With SD, HD, DD being either positive or negative giving 8 branches, and also how healthy the ship is, are the shields up, hull > or < minimumHull.

Reply:
Looks good!